packages/grafana-prometheus/src/querybuilder/components/PromQueryBuilderOptions.tsx

```diff
@@ -112,7 +112,7 @@ export const PromQueryBuilderOptions = React.memo<PromQueryBuilderOptionsProps>(
                 <EditorSwitch
                   value={query.exemplar || false}
                   onChange={onExemplarChange}
-                  id={selectors.components.DataSource.Prometheus.queryEditor.exemplars}
+                  data-test-id="prometheus-exemplars"
```

Comment:
`data-testid` is what's used everywhere else FYI.

`data-testid={selectors.components.DataSource.Prometheus.queryEditor.exemplars}`

would then allow you to write `selectors.components.DataSource.Prometheus.queryEditor.exemplars().scrollIntoView().should('exist')` in your test.

Reply:
I know and I wanted to use it but the component is overriding it. I cannot select the component with cypress.

Reply:
[I had this issue last spring with the select component](https://github.com/grafana/grafana/issues/79861#issue-2056665875). I think the EditorSwitch is similar. There need to be some work done to pass the `data-testid` from the EditorSwitch in grafana/experiment to the Switch in grafana/ui, all the way down, so that we can call
`selectors.components.DataSource.Prometheus.queryEditor.exemplars().scrollIntoView().should('exist')`